modules/archetypes/variables.tf

```diff
@@ -36,7 +36,7 @@ variable "parameters" {
 }
 
 variable "enforcement_mode" {
-  type        = map(string)
+  type        = map(bool)
```

Comment:
@matt-FFFFFF I'm not sure this will fix it, as the module is being passed what looks like a `map(string)`. https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/blob/6b87560d83653de33900903ee8d5b369737b845a/main.tf#L19

Reply:
Actually, looking deeper into the code, it is an `any`, but the docs indicate is should be a `map(bool)`. Apologies for the confusion!

Reply:
Here: https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/blob/6b87560d83653de33900903ee8d5b369737b845a/variables.tf#L452

Reply:
Realised what's wrong #758 needs merging! Can you have a look and review please?

Reply:
Sorry #762